components/table/table-col-sort-button.js

```diff
@@ -84,17 +95,57 @@ export class TableColSortButton extends FocusMixin(LitElement) {
 		super();
 		this.nosort = false;
 		this.desc = false;
+
+		this._describedById = getUniqueId();
 	}
 
 	static get focusElementSelector() {
 		return 'button';
 	}
 
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+
+		if (!this.dataType) {
+			console.warn('d2l-table-col-sort-button: data-type attribute does not have a value.');
+		}
+	}
+
 	render() {
+		const buttonDescription = this.nosort ? this.localize('components.table-col-sort-button.addSortOrder') : this.localize('components.table-col-sort-button.changeSortOrder');
+		const buttonTitle = this.nosort
+			? undefined
+			: this.localize('components.table-col-sort-button.title', {
+				dataType: this.dataType,
+				direction: this.desc ? 'desc' : undefined
+			});
```

Comment:
I used the nested template on the Daylight localization sandbox to generate the lang term. It's a bit debatable on if we should use that one vs just having the code specify which lang term to use based on `data-type` and direction, but we would end up with 8 additional lang terms instead of one. Let me know thoughts on this.